### PR TITLE
Environment Drivers Refactor + Documentation

### PR DESCRIPTION
**Pull request by devmitch**
Since #334 abstracted `ExtrinsicDrivers` into its own class, we can now clean up the old driver API in `Environment` and replace all the methods with an exposed `ExtrinsicDrivers` object. #334 also included a smarter way to set up drivers, which is also included in this PR:
Default (rainfall = 1):
```r
env <- make_environment("FF16w")
```
Constant:
```r
env <- make_environment("FF16w", rainfall=3.14)
```
Variable:
```r
x_pts <- seq(0, 200, 1)
rain <- list(
  x = x_pts,
  y = 1 + sin(x_pts)
)
env <- make_environment("FF16w", rainfall=rain)
```
Evaluation is now done like this:
```r
env$extrinsic_drivers$evaluate("rainfall", 5.634)
env$extrinsic_drivers$evaluate_range("rainfall", c(5, 1.2, 78.345))
```

These examples have also been included in a new vignette at `vignettes/extrinsic_drivers.Rmd`. It is only a draft as of now. I can't seem to knit R code or embed images (other vignettes use a `figure/` directory, which I can't seem to find...). I would appreciate your input on the documentation!

### CODE REVIEW

**Review by aornugent — APPROVED** [review]

**Comment by aornugent** [comment]
This looks great @devmitch! I forget exactly the workflow for rebuilding the website, but can knit your vignette successfully.

**Comment by dfalster** [comment]
Hi @devmitch 

Vignette is very good! I just added some minor changes. You can test rendering using 

```
rmarkdown::render("vignettes//strategy_FF16.Rmd")
```

I've used another vignette above, as currently this branch isn't compiling. E.g. see 

https://github.com/traitecoevo/plant/runs/7160778238?check_suite_focus=true

Can you follow up on the compilation?

**Comment by dfalster** [comment]
I'll post a separate issue about updating the vignettes in general. Bunch are out of date.

**Comment by dfalster** [comment]
Also, @devmitch 

How does this branch relate to https://github.com/traitecoevo/plant/tree/doc-updates?

**Review by dfalster — CHANGES_REQUESTED** [review]
Looking very good. 

However, this branch is currently not compiling for me, or passing GH actions.

**Comment by devmitch** [comment]
> Also, @devmitch
> 
> How does this branch relate to https://github.com/traitecoevo/plant/tree/doc-updates?

Pretty sure that branch contains an outdated and incomplete version of the vignette I have pushed on this branch.

> Looking very good.
> 
> However, this branch is currently not compiling for me, or passing GH actions.

Hopefully everything should be compiling now, I'm going to attempt the get_state changes on another branch/PR just to separate concerns.

**Review by dfalster — APPROVED** [review]
Nice work @devmitch

**Comment by dfalster** [comment]
Feel free to merge

File: vignettes/extrinsic_drivers.Rmd [hunk 1]
```diff
@@ -77,6 +77,7 @@ birth_rates <- list(
 p1 <- expand_parameters(lmas, p0, FF16_hyperpar, FALSE, birth_rates)
 
 # set up env and ctrl...
+ctrl <- Control()
```
Comment (aornugent) [inline vignettes/extrinsic_drivers.Rmd:80]: Maybe try `scm_base_control` here.

**Pull request merged**